functions/form-payment-reconciliation/index.js

```diff
@@ -0,0 +1,291 @@
+/**
+ * Main file that runs payment reconciliation CRON job.
+ * Required env vars:
+ * - AWS_REGION: the aws region which the
+ * - SSM_ENV_SITE_NAME: ['prod', 'staging', 'staging-alt', 'staging-alt2', 'uat']
+ *
+ * Required parameters in parameter store `<SSM_ENV_SITE_NAME>-cron-payment`:
+ * - CRON_PAYMENT_API_SECRET: the shared secret to validate protected routes in FormSG API
+ * - SLACK_API_SECRET: the API key for posting reports to slack
+ */
+
+const { SSMClient, GetParameterCommand } = require('@aws-sdk/client-ssm')
+const { generateLinkToCloudwatch, getPostToSlack } = require('./src/utils')
+
+const AWS_REGION = process.env.AWS_REGION
+
+const ENV_SITE_NAME = process.env.SSM_ENV_SITE_NAME
+const CRON_API_PREFIX = `https://${
+  ENV_SITE_NAME === 'prod' ? '' : `${ENV_SITE_NAME}.`
+}form.gov.sg/api/v3/payments/reconcile/`
+
+const PARAMETER_STORE_NAME = `${ENV_SITE_NAME}-cron-payment`
+const CRON_PAYMENT_API_SECRET_KEY = 'CRON_PAYMENT_API_SECRET'
+const CRON_PAYMENT_SLACK_SECRET_KEY = 'SLACK_API_SECRET'
+
+const API_AUTH_HEADER = 'x-formsg-cron-payment-secret'
+
+const MAX_AGE_HRS_EVENTS = 7
+
+/**
+ * Helper function to obtain secrets map from parameter store.
+ */
+const getSSMSecrets = () => {
+  // Regex used to match key-value pairs from parameter store.
+  // Makes use of capture groups to obtain the key and value.
+  const KEY_VALUE_PAIR_REGEX = /^([^\s=]+)\s*=\s*(\S+)$/
+
+  const awsSsmClient = new SSMClient({ region: AWS_REGION })
+  const input = {
+    Name: PARAMETER_STORE_NAME,
+    WithDecryption: true,
+  }
+  const command = new GetParameterCommand(input)
+  return awsSsmClient.send(command).then((res) => {
+    const parametersAsString = res.Parameter.Value
+    const parameters = parametersAsString
+      .split(/[\r\n]+/)
+      .map((line) => line.trim())
+      .filter((line) => !!line) // Filter out empty lines
+    return parameters.reduce((parameterMap, line) => {
+      const match = line.match(KEY_VALUE_PAIR_REGEX)
+      if (match && match.length === 3) parameterMap[match[1]] = match[2]
```

Comment:
/nit: The length check for 3 is not needed. If `match` is truthy, it is guaranteed to have length 3 (thank you regexes!)

... Hmm, but is it because typescript doesn't know about it? 😅 

```suggestion
      if (match) parameterMap[match[1]] = match[2]
```

Reply:
this one is purely my preference to have an explicit guarantee that the array index won't explode rather than relying on the behavior of `.match` 😅

Reply:
Ah I see. But hmm, you have to be careful when preference goes in the way of readability (somewhat) 🤔 : a sanity/defensive check implies something could happen, and a careful reader must expand some mental load to understand what. But in this case, the behaviour of the `.match()` is guaranteed.

BTW, if you find it clearer, you could use named groups:

```javascript
const KEY_VALUE_PAIR_REGEX = /^(?<key>[^\s=]+)\s*=\s*(?<value>\S+)$/

const match = line.match(KEY_VALUE_PAIR_REGEX)

if (match) parameterMap[match.groups.key] = match.groups.value
```

Where similarly, `match` being truthy guarantees that all defined groups have a value (yay regexes!)